redis-hbo-provider/src/main/java/com/facebook/presto/statistic/RedisProviderPlugin.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.statistic;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.spi.Plugin;
+import com.facebook.presto.spi.statistics.EmptyPlanStatisticsProvider;
+import com.facebook.presto.spi.statistics.HistoryBasedPlanStatisticsProvider;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Injector;
+import io.lettuce.core.AbstractRedisClient;
+
+import java.util.Map;
+
+import static com.facebook.presto.statistic.PropertiesUtil.loadAndProcessProperties;
+
+public class RedisProviderPlugin
+        implements Plugin, AutoCloseable
+{
+    private static final Logger log = Logger.get(RedisProviderPlugin.class);
+    private final Map<String, String> propertyMap;
+
+    private Injector injector;
+
+    @VisibleForTesting
+    public RedisProviderPlugin(String configPath)
+    {
+        // This plugin cannot be loaded like other plugins because this is not a connector
+        Map<String, String> propertyMap;
+        try {
+            propertyMap = loadAndProcessProperties(configPath);
+        }
+        catch (Exception e) {
+            propertyMap = null;
+            log.info("Redis Provider Plugin was not loaded since config was not loaded properly");
+        }
+        this.propertyMap = propertyMap;
+    }
+
+    public RedisProviderPlugin()
+    {
+        this(RedisProviderConfig.REDIS_PROPERTIES_PATH);
+    }
+
+    public Iterable<HistoryBasedPlanStatisticsProvider> getHistoryBasedPlanStatisticsProviders()
+    {
+        if (this.propertyMap == null) {
+            log.info("Redis Provider Plugin returned an empty instance");
+            return ImmutableList.of(EmptyPlanStatisticsProvider.getInstance());
+        }
+        if (!(propertyMap.getOrDefault(RedisProviderConfig.COORDINATOR_PROPERTY_NAME, "false").equals("true") &&
+                propertyMap.getOrDefault(RedisProviderConfig.HBO_PROVIDER_ENABLED_NAME, "false").equals("true"))) {
+            return ImmutableList.of(EmptyPlanStatisticsProvider.getInstance());
+        }
```

Comment:
I agree yes I can do this and that would remove the constants and their use outside the config. 
 To confirm, we'd still have to manually load the .properties file from the storage because airlift lacks the functionality to do so natively. In other connectors we do the initialization through [ConnectorFactory](https://github.com/prestodb/presto/blob/cea8ab195d67cc798a0405b6c7a21edcd4aa9722/presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorFactory.java#L26) which gives the propertyMap directly. For plugins this is not supported. 
Hence I've modeled the loading process after [PrestoSparkLauncherCommand](https://github.com/prestodb/presto/blob/dd5ae3bd7a572f1c66d25b303af49fb3c2ab0802/presto-spark-launcher/src/main/java/com/facebook/presto/spark/launcher/PrestoSparkLauncherCommand.java#L68) .
Not sure if there is any other way

Reply:
> // double check: why do we want to check if coordinator == true? 
We do not want the workers to maintain an active connection to redis since only the co-ordinator is going to use them

Reply:
We also make use of a number of @Provides annotations within 
 [InjectorFactory](https://github.com/prestodb/presto/blob/73687d27a5495c2ed282bee1868c759180a7fe0d/redis-hbo-provider/src/main/java/com/facebook/presto/statistic/RedisProviderInjectorFactory.java). Managing them cleanly is something I'm unsure about. Additionally, since we don't need any classes if these configurations are set to false, we'll need to implement this check in the initialization of all classes as well as in each provision. An alternative approach would be to construct the configuration initially in a separate bootstrap instance and then utilize those configuration variables. Alternatively, we could simply leave the code as it is if we must load the file anyway.

Reply:
I see the problem.  Let's keep it as is.